fix .so file name error on linux and macos

handong0123 opened the pull request:
I have reported an error when using tensorflow1.14 on linux. The program prompts that libtensorflow_framework.so.1 cannot be found. It is found that only libtensorflow_framework.so is decompressed under /tmp, so I modified the NativeLibrary class and corrected the name when releasing the resource.

run on linux error：
java.lang.UnsatisfiedLinkError: /tmp/tensorflow_native_libraries-1562914806051-0/libtensorflow_jni.so: libtensorflow_framework.so.1: cannot open shared object file: No such file or directory

run on macos error：
Exception in thread "main" java.lang.UnsatisfiedLinkError: /private/var/folders/0c/nz0823ps085b_1zcq6l8ry441kqw10/T/tensorflow_native_libraries-1562915587706-0/libtensorflow_jni.dylib: dlopen(/private/var/folders/0c/nz0823ps085b_1zcq6l8ry441kqw10/T/tensorflow_native_libraries-1562915587706-0/libtensorflow_jni.dylib, 1): Library not loaded: @rpath/libtensorflow_framework.1.dylib
  Referenced from: /private/var/folders/0c/nz0823ps085b_1zcq6l8ry441kqw10/T/tensorflow_native_libraries-1562915587706-0/libtensorflow_jni.dylib
  Reason: image not found

handong0123 committed:
fix .so file name error on linux and macos

handong0123 committed:
Update NativeLibrary.java

allenlavoie reviewed (commented):
Can you modify the getVersionedLibraryName method instead of fixing its incorrect output afterwords?

And do you know what the issue is there? Could be something like getMajorVersionNumber returning 2 but the shared object filenames haven't been updated yet. In which case we should update the shared objects, or just fix the version number at 1 for now since the Java API hasn't changed for TF 2.x.

18867664853 commented:
I will try to check the method and resolve this problem, thanks

18867664853 commented:
> 您可以修改getVersionedLibraryName方法，而不是修复其不正确的输出后缀吗？
> 
> 而且您知道存在什么问题吗？可能类似于getMajorVersionNumber返回2，但共享对象文件名尚未更新。在这种情况下，我们应该更新共享对象，或者暂时将版本号固定为1，因为TF 2.x的Java API并未更改。


I found the problem, getMajorVersionNumber() uses the package's getImplementVersion(), which returns the implementation version of the MANIFEST.MF in the jar package, but tensorflow does not declare the pointer when the jar is called, resulting in an empty return, so Make the suffix of the .so file incorrect.

I found that even if MANIFEST.MF contains the Implementation-Version field, getImplementVersion() does not return the version number correctly. I am at https://stackoverflow.com/questions/33553570/how-to-read-meta-data-from-manifest -file found a way to correctly read the Implementation-Version.

So I think we can temporarily fix the version number to 1, and then tf2.x declares the Implementation-Version.

handong0123 committed:
Temporarily fixed version number is 1

I think we can temporarily fix the version number to 1, and then tf2.x declares the Implementation-Version.

allenlavoie commented on tensorflow/java/src/main/java/org/tensorflow/NativeLibrary.java:
```diff
@@ -139,9 +133,9 @@ private static boolean resourceExists(String baseName) {
-    if (resourceExists(libFilename)) {
-      return libFilename;
-    }
+    // if (resourceExists(libFilename)) {
```
Comment-out left over from debugging?

18867664853 replied:
正在使用发音
This is the original code, but because there is libtensorfow_framework.so in the jar package, which causes the later logic to fail, I temporarily commented out the code.

allenlavoie replied:
I'm confused then. If the unversioned shared object exists, why isn't it used? This logic did work before we added version numbers to shared objects.

18867664853 replied:
There are libtensorflow_framework.so.so and libtensorflow_framework.so.1. libtensorflow_jni.so seems to depend on libtensorflow_framework.so.1, but now the logic detects that libtensorflow_framework.so is over. Does he seem to have a problem?

allenlavoie replied:
Thanks for the explanation. Since we'd fall back to the unversioned lib anyway I think we can just delete this snippet rather than commenting out.

18867664853 replied:
I think that's a good suggestion. Now I'll delete the logic of judgment.

allenlavoie commented on tensorflow/java/src/main/java/org/tensorflow/NativeLibrary.java:
```diff
@@ -180,11 +174,13 @@ private static String getVersionedLibraryName(String libFilename) {
    * determined.
    */
   private static String getMajorVersionNumber() {
+    // getImplementationVersion() retrun null. 
```
Can return null? Or always returns null right now?

18867664853 replied:
always returns null right now.

allenlavoie reviewed (commented):
Minor nits; the approach seems fine.

handong0123 committed:
Update NativeLibrary.java

delete unversioned lib judge
